### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by nathanshelly**
Move additional methods out of Game to GameUtility
Add additional GameUtility tests
Fix issue where back-end would send board to front-end up to three times on turn change

### CODE REVIEW

File: src/Game.ts [hunk 1]
```diff
@@ -179,16 +162,16 @@ export class Game {
...
 
 	switchActiveTeam(): void {
     this.currTeam = this.currTeam === Team.red ? Team.blue : Team.red;
 		this.turn = Turn.spy;
-		var spymasters = this.findSpymasters();
+		
 		Broadcaster.switchActiveTeam(this.agents, this.currTeam, this.turn);
-		Broadcaster.promptForClue(spymasters[this.currTeam]);
+		
+		let currTeamSpy: Spymaster[] = gu.getByTeam(gu.getSpymasters(this.agents), this.currTeam)
```
Comment (vkoves) [inline src/Game.ts]: I'd change this to `currTeamMaster` or `currTeamSpymaster`, as spy could be confused with the agents that are normal players.

File: src/Game.ts [hunk 2]
```diff
@@ -271,9 +256,11 @@ export class Game {
 
 				this.checkGuess(currSelection as number);
 
-				this.findOperatives().filter(op => op.team === sop.team).forEach(
-					innerOp => this.toggleCard(innerOp, currSelection)
-				)
+				gu.getByTeam(gu.getOperatives(this.agents), sop.team).forEach(op => 
+						this.toggleCard(op, currSelection))
```
Comment (vkoves) [inline src/Game.ts]: This filter should be closed by a semicolon, and could be moved to one line. I'd also suggest adding an inline comment explaining what this does, as I'm not sure.

File: src/GameUtility.ts [hunk 3]
```diff
@@ -20,7 +22,28 @@ export module GameUtility {
 		return team === Team.red ? Team.blue : Team.red;
 	}
 
-	export function getTeamOps(allOps: Operative[], team: Team): Operative[] {
-		return allOps.filter(op => op.team === team);
+	export function getStartTeam(): Team {
+  	return Math.round(Math.random()) ? Team.red : Team.blue ;
```
Comment (vkoves) [inline src/GameUtility.ts]: This should be indented.

File: test/rule_enforcer.test.ts [hunk 4]
```diff
@@ -154,25 +155,25 @@ describe("Filename: rules_enforcer.test.ts:\n\nRules Enforcer", () => {
 		let agents: Agent[] = [red_p_one, red_p_two, red_p_three, blue_p_one, blue_p_two];
 
 		// TODO: Use utility method for getting ops
-		let ops: Operative[] = agents.filter(agent => agent.role === Turn.op) as Operative[];
+		let ops: Teams<Operative> = gu.getOperatives(agents);
 		let select_card = new Card("select");
 		let decoy_card = new Card("decoy");
 
-		expect(re.canSubmitGuess(ops, Team.red)).to.be.false;
+		expect(re.canSubmitGuess(gu.getByTeam(ops, Team.red))).to.be.false;
```
Comment (vkoves) [inline test/rule_enforcer.test.ts:162]: I'd generally avoid two letter variables like `re` or `gu`. Can you change those to be `RuleEnforcer` and `Game Utility` instead? That seems much clearer and removes the need for doing the `import ... as`. This should be changed across the board.

**Review by vkoves — COMMENTED** [review]
Added a few comments for code styling and clarifying variables.

File: test/rule_enforcer.test.ts (hunk 4 above)
Comment (nathanshelly) [reply test/rule_enforcer.test.ts:162]: I tend to agree in general but value the brevity of renaming imports in situations like this (see Lodash imported as _)

**Review by vkoves — APPROVED** [review]
Looks good!

**Pull request merged**